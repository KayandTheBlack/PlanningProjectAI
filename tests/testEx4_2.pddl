(define (problem autogen)
  (:domain worldsEndAndBeyond)

  (:objects
    Origin cityA cityB cityC cityD cityE cityF cityG cityH - city
    hotelA_A hotelA_B hotelB_A hotelB_B hotelC_A hotelC_B hotelD_A hotelD_B hotelE_A hotelE_B hotelF_A hotelF_B hotelG_A hotelG_B hotelH_A hotelH_B - hotel
  )

  (:init
    (currentLocation Origin)

    (hotelAt hotelA_A cityA)
    (hotelAt hotelA_B cityA)
    (hotelAt hotelB_A cityB)
    (hotelAt hotelB_B cityB)
    (hotelAt hotelC_A cityC)
    (hotelAt hotelC_B cityC)
    (hotelAt hotelD_A cityD)
    (hotelAt hotelD_B cityD)
    (hotelAt hotelE_A cityE)
    (hotelAt hotelE_B cityE)
    (hotelAt hotelF_A cityF)
    (hotelAt hotelF_B cityF)
    (hotelAt hotelG_A cityG)
    (hotelAt hotelG_B cityG)
    (hotelAt hotelH_A cityH)
    (hotelAt hotelH_B cityH)

    (flight Origin cityA)
    (flight Origin cityB)
    (flight Origin cityC)
    (flight Origin cityD)
    (flight Origin cityE)
    (flight Origin cityF)
    (flight Origin cityG)
    (flight Origin cityH)
    (flight cityA cityB)
    (flight cityA cityD)
    (flight cityA cityE)
    (flight cityA cityF)
    (flight cityB cityA)
    (flight cityB cityD)
    (flight cityB cityG)
    (flight cityB cityH)
    (flight cityC cityA)
    (flight cityC cityB)
    (flight cityC cityF)
    (flight cityC cityG)
    (flight cityD cityB)
    (flight cityD cityC)
    (flight cityD cityG)
    (flight cityD cityH)
    (flight cityE cityA)
    (flight cityE cityC)
    (flight cityE cityF)
    (flight cityE cityG)
    (flight cityF cityA)
    (flight cityF cityC)
    (flight cityF cityE)
    (flight cityF cityH)
    (flight cityG cityA)
    (flight cityG cityC)
    (flight cityG cityE)
    (flight cityG cityF)
    (flight cityH cityA)
    (flight cityH cityB)
    (flight cityH cityC)
    (flight cityH cityG)

    (= (citiesVisited) 0)
    (= (totalDays) 0)
    (= (minDaysInCity) 1)
    (= (maxDaysInCity) 2)

    (= (stayDuration Origin) 0)
    (= (stayDuration cityA) 0)
    (= (stayDuration cityB) 0)
    (= (stayDuration cityC) 0)
    (= (stayDuration cityD) 0)
    (= (stayDuration cityE) 0)
    (= (stayDuration cityF) 0)
    (= (stayDuration cityG) 0)
    (= (stayDuration cityH) 0)
    (= (cityInterest Origin) 0)
    (= (cityInterest cityA) 3)
    (= (cityInterest cityB) 1)
    (= (cityInterest cityC) 1)
    (= (cityInterest cityD) 1)
    (= (cityInterest cityE) 2)
    (= (cityInterest cityF) 2)
    (= (cityInterest cityG) 3)
    (= (cityInterest cityH) 3)
    (= (totalInterest) 0)
    (= (totalcost) 0)
    (= (hotelcost hotelA_A) 233)
    (= (hotelcost hotelA_B) 117)
    (= (hotelcost hotelB_A) 147)
    (= (hotelcost hotelB_B) 231)
    (= (hotelcost hotelC_A) 177)
    (= (hotelcost hotelC_B) 50)
    (= (hotelcost hotelD_A) 177)
    (= (hotelcost hotelD_B) 90)
    (= (hotelcost hotelE_A) 176)
    (= (hotelcost hotelE_B) 239)
    (= (hotelcost hotelF_A) 226)
    (= (hotelcost hotelF_B) 100)
    (= (hotelcost hotelG_A) 159)
    (= (hotelcost hotelG_B) 202)
    (= (hotelcost hotelH_A) 240)
    (= (hotelcost hotelH_B) 201)

    (= (flightcost Origin cityA) 0)
    (= (flightcost Origin cityB) 0)
    (= (flightcost Origin cityC) 0)
    (= (flightcost Origin cityD) 0)
    (= (flightcost Origin cityE) 0)
    (= (flightcost Origin cityF) 0)
    (= (flightcost Origin cityG) 0)
    (= (flightcost Origin cityH) 0)
    (= (flightcost cityA cityB) 189)
    (= (flightcost cityA cityD) 149)
    (= (flightcost cityA cityE) 185)
    (= (flightcost cityA cityF) 164)
    (= (flightcost cityB cityA) 159)
    (= (flightcost cityB cityD) 142)
    (= (flightcost cityB cityG) 104)
    (= (flightcost cityB cityH) 152)
    (= (flightcost cityC cityA) 169)
    (= (flightcost cityC cityB) 152)
    (= (flightcost cityC cityF) 184)
    (= (flightcost cityC cityG) 172)
    (= (flightcost cityD cityB) 130)
    (= (flightcost cityD cityC) 189)
    (= (flightcost cityD cityG) 139)
    (= (flightcost cityD cityH) 113)
    (= (flightcost cityE cityA) 156)
    (= (flightcost cityE cityC) 136)
    (= (flightcost cityE cityF) 146)
    (= (flightcost cityE cityG) 136)
    (= (flightcost cityF cityA) 188)
    (= (flightcost cityF cityC) 174)
    (= (flightcost cityF cityE) 176)
    (= (flightcost cityF cityH) 114)
    (= (flightcost cityG cityA) 115)
    (= (flightcost cityG cityC) 152)
    (= (flightcost cityG cityE) 164)
    (= (flightcost cityG cityF) 124)
    (= (flightcost cityH cityA) 157)
    (= (flightcost cityH cityB) 155)
    (= (flightcost cityH cityC) 128)
    (= (flightcost cityH cityG) 146)

  )

  (:goal
    (and
      (>= (citiesVisited) 4)
      (>= (totalDays) 10)
      (>= (totalcost) 1500)
      (<= (totalcost) 2000)
    )
  )

  (:metric 
    minimize
      (+ (* 350 (totalInterest)) (totalcost))
  )

)
