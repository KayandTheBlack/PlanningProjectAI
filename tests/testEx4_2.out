
ff: parsing domain file
domain 'WORLDSENDANDBEYOND' defined
 ... done.
ff: parsing problem file
problem 'AUTOGEN' defined
 ... done.


metric established (normalized to minimize): ((350.00*[RF1](TOTALINTEREST)1.00*[RF0](TOTALCOST)) - () + 0.00)

checking for cyclic := effects --- OK.

ff: search configuration is  best-first on 1*g(s) + 5*h(s) where
    metric is ((350.00*[RF1](TOTALINTEREST)1.00*[RF0](TOTALCOST)) - () + 0.00)

advancing to distance:   10
                          9
                          8
                          7
                          6
                          5
                          4
                          3
                          2
                          1
                          0

ff: found legal plan as follows

step    0: FLYANDSTAY ORIGIN CITYA HOTELA_B
        1: FLYANDSTAY CITYA CITYE HOTELE_A
        2: FLYANDSTAY CITYE CITYG HOTELG_A
        3: INCREASESTAY CITYE HOTELE_A
        4: INCREASESTAY CITYG HOTELG_A
        5: INCREASESTAY CITYA HOTELA_B
        6: FLYANDSTAY CITYG CITYC HOTELC_B
        7: FLYANDSTAY CITYC CITYB HOTELB_A
        8: INCREASESTAY CITYC HOTELC_B
        9: INCREASESTAY CITYB HOTELB_A
     

time spent:    0.00 seconds instantiating 96 easy, 0 hard action templates
               0.00 seconds reachability analysis, yielding 42 facts and 96 actions
               0.00 seconds creating final representation with 41 relevant facts, 22 relevant fluents
               0.00 seconds computing LNF
               0.00 seconds building connectivity graph
               0.01 seconds searching, evaluating 72 states, to a max depth of 0
               0.01 seconds total time

